functions/Test-DbaLoginPassword.ps1

```diff
@@ -0,0 +1,113 @@
+            Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+        .NOTES
+            Author: Peter Samuelsson
+
+            dWebsite: https://dbatools.io
+            Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+            License: MIT https://opensource.org/licenses/MIT
+
+        .LINK
+            https://dbatools.io/Test-DbaLoginPassword
+
+        .EXAMPLE
+            Test-DbaLoginPassword -SqlInstance Dev01
+
+            Test all SQL logins that the password is null or same as username on SQL server instance Dev01
+
+        .EXAMPLE
+            Test-DbaLoginPassword -SqlInstance Dev01 -Dictionary Test1,test2
+
+            Test all SQL logins that the password is null, same as username or Test1,Test2 on SQL server instance Dev0
+    #>
+
+    [CmdletBinding()]
+    Param (
+        [parameter(Position = 0, Mandatory = $true, ValueFromPipeline = $true)]
+        [Alias("ServerInstance", "SqlServer", "SqlServers")]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [String[]]$Dictionary,
+        [switch]$EnableException
+    )
+
+    begin {
+        $CheckPasses = "''", "'@@Name'"
+        if ($Dictionary) {
+            $Dictionary | ForEach-Object {$CheckPasses += "'" + $psitem + "'" }
+        }
+
+        foreach ($CheckPass in $CheckPasses) {
+            if ($CheckPasses.IndexOf($CheckPass) -eq 0) {
+                $checks = "SELECT " + $CheckPass
+            } else {
+                $checks += "
+        UNION SELECT " + $CheckPass
+            }
+        }
+
+        $sql = "DECLARE @WeakPwdList TABLE(WeakPwd NVARCHAR(255))
+            --Define weak password list
+            --Use @@Name if users password contain their name
+            INSERT INTO @WeakPwdList(WeakPwd)
+            $checks
+
+            SELECT SERVERPROPERTY('MachineName') AS [ComputerName],
+                SERVERPROPERTY('InstanceName') AS [Instance],
+                SERVERPROPERTY('ServerName') AS [SqlInstance],
+                SysLogins.name as SqlLogin,
+                REPLACE(WeakPassword.WeakPwd,'@@Name',SysLogins.name) As [Password],
+                SysLogins.is_disabled as Disabled,
+                SysLogins.create_date as CreatedDate,
+                SysLogins.modify_date as ModifiedDate,
+                SysLogins.default_database_name as DefaultDatabase
+            FROM sys.sql_logins SysLogins
+            INNER JOIN @WeakPwdList WeakPassword ON (PWDCOMPARE(WeakPassword.WeakPwd, password_hash) = 1
+                OR PWDCOMPARE(REPLACE(WeakPassword.WeakPwd,'@@Name',SysLogins.name),password_hash) = 1)"
+    }
+    process {
+        foreach ($instance in $SqlInstance) {
+            try {
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $sqlcredential
```

Comment:
This needs to include -MinimumVersion 10 unless you tested and know the view in use works against 2005?

Reply:
Ahhh didn't know that. Change the code